consensus/polybft/fsm.go

```diff
@@ -648,3 +662,35 @@ func createStateTransactionWithData(target types.Address, inputData []byte) *typ
 
 	return tx
 }
+
+// getExitEventsFromReceipts parses logs from receipts to find exit events
+func getExitEventsFromReceipts(epoch, block uint64, receipts []*types.Receipt) ([]*ExitEvent, error) {
+	events := make([]*ExitEvent, 0)
+
+	for i := 0; i < len(receipts); i++ {
+		if len(receipts[i].Logs) == 0 {
+			continue
+		}
+
+		for j := 0; j < len(receipts[i].Logs); j++ {
+			event, err := decodeExitEvent(convertLog(receipts[i].Logs[j]), epoch, block)
```

Comment:
The final thing there. We need to be sure that the event was from our exit contract. Do you have this check somewhere? I havent managed to find.

Reply:
This should do the trick: https://github.com/0xPolygon/polygon-edge/pull/852/files#diff-514a0d26a7bee06b50bd0d4d72613aa918eb7ae096be6ea973ea6e2680d4dd75R132. Hopefully there wouldn't be the same function signature on some of the other smart contracts? :slightly_smiling_face:

Reply:
Let's add that check of originating address once we implement deploying of checkpoint contracts. In my understanding, contracts will be deployed as part of genesis, and we will have a known (fixed) address, which we can use here to check if the address match. I'll write this down so we do not forget.

Reply:
> event signature emitted from multiple contracts

This could be an attack vector on the exit contract if we do not hardcode the address. Anyone could create the same event and emit it, it is not per-se about the chances of generating the same event signature.

Reply:
Created a separate issue for this address check, once we implement deployment of the exit contract:

https://polygon.atlassian.net/browse/EVM-162